packages/shared/src/metadata-manager.ts

```diff
@@ -0,0 +1,127 @@
+import {
+  MetadataEntry,
+  EncryptedMetadataManager,
+  MetadataManager,
+  SettingsManager,
+} from "@saleor/app-sdk/settings-manager";
+import { Client, gql } from "urql";
+
+const UpdateAppMetadataMutation = gql`
+  mutation UpdateAppMetadata($id: ID!, $input: [MetadataInput!]!) {
+    updatePrivateMetadata(id: $id, input: $input) {
+      item {
+        privateMetadata {
+          key
+          value
+        }
+      }
+    }
+  }
+`;
+
+const FetchAppDetailsQuery = gql`
+  query FetchAppDetails {
+    app {
+      id
+      privateMetadata {
+        key
+        value
+      }
+    }
+  }
+`;
+
+const DeletePrivateMetadataMutation = gql`
+  mutation RemovePrivateMetadata($id: ID!, $keys: [String!]!) {
+    deletePrivateMetadata(id: $id, keys: $keys) {
+      errors {
+        message
+      }
+    }
+  }
+`;
+
+/**
+ * To avoid a graphql-schema build step, manually set types for the queries and mutations.
+ */
+type FetchAppPrivateMetadataQuery = {
+  __typename?: "Query";
+  app?: {
+    __typename?: "App";
+    id: string;
+    privateMetadata: Array<{ __typename?: "MetadataItem"; key: string; value: string }>;
+  } | null;
+};
+
+type UpdateAppPrivateMetadataMutation = {
+  __typename?: "Mutation";
+  updatePrivateMetadata?: {
+    __typename?: "UpdatePrivateMetadata";
+    item?: {
+      __typename?: "App";
+      privateMetadata: Array<{ __typename?: "MetadataItem"; key: string; value: string }>;
+    } | null;
+  } | null;
+};
+
+export type MetadataManagerGraphqlClient = Pick<Client, "mutation" | "query">;
+
+async function fetchAllPrivateMetadata(
```

Comment:
**Question[^1][^2]**:
What do you think about exporting this function too? I'm using these queries not only in App metadata, but for Category too (products feed app)

[^1]: ✅ Not blocking the review. 
[^2]: I just wanted to know more context, or spotted a cool solution I would like to know more about :)

Reply:
I dont tthink we should mix it here. I mean, I will export it and you will create a coupling between metadata manager and categories, which was not meant to happen.

I think its ok to extract "common" queries to shared, but then lets create a separate package for it and explicitly design it to work with categories etc (or anything)